3rd PR: Add supplementary features to the CUBRID plugin

longhaseng52 opened the pull request:
- Display list of indexes
- Display list of serials
- Display list of synonyms
- Display list of triggers
- Display list of functions & procedures
- Display DDL of view
- Create & edit table using UI

longhaseng52 committed:
Display list of indexes

longhaseng52 committed:
Display list of serials

longhaseng52 committed:
Display list of synonyms

longhaseng52 committed:
Display list of triggers

longhaseng52 committed:
Display list of procedures and functions

longhaseng52 committed:
Display DDL of view

longhaseng52 committed:
Create and edit table using UI

hwany7seo commented on plugins/org.jkiss.dbeaver.ext.cubrid/src/org/jkiss/dbeaver/ext/cubrid/model/CubridTableColumn.java:
```diff
@@ -0,0 +1,118 @@
+        	long columnSize = 0;
+            typeName = JDBCUtils.safeGetString(dbResult, "Type");
+            if (typeName != null && typeName.contains("(")) {
+                columnSize = Long.parseLong(typeName.replaceAll("[^0-9]", ""));
```
If the value is empty, an exception may occur.
e.g) ENUM('Y', 'N')

longhaseng52 replied:
Well noted, I will fix this issues. However, currently This PR is covered data type following Generic data type. So, Enum type will not display in UI, can only create using script.

hwany7seo replied:
We are planning to modify the JDBC Driver and
After modification, it appears as below.
![image](https://github.com/longhaseng52/dbeaver/assets/68524199/5ea62290-590f-4e9d-9b44-e20e2482a2f4)

Additionally, an exception may appear when retrieving data that has already been entered.

longhaseng52 replied:
I've fixed the error when retrieving data in the last commit.

hwany7seo commented:
Datatype cannot be selected. It's a textbox.
![image](https://github.com/longhaseng52/dbeaver/assets/68524199/101393be-c591-4bc8-8d71-26aaf0763581)
![image](https://github.com/longhaseng52/dbeaver/assets/68524199/b12772c7-afd9-42e7-84f1-1e20c273221a)

longhaseng52 committed:
Fixed data type from textbox to select as option

longhaseng52 committed:
Removed unused file CubridInformation.java

longhaseng52 committed:
Fixed error exception when loading specific data type

hwany7seo commented:
Please include the following as desired by DBeaver engineers.

- Place { at the same line where the method is declared
- Please place all private methods at the bottom of the class

longhaseng52 commented:
> Please include the following as desired by DBeaver engineers.
> 
> * Place { at the same line where the method is declared
> * Please place all private methods at the bottom of the class

Well noted. I will change it at last.

longhaseng52 commented:
Dear @hwany7seo,

QA team just found out that the column order is not matching with the design. It is because I follow the column order based on Generic. So, Should I change it following our design?

![image](https://github.com/longhaseng52/dbeaver/assets/123133969/d5239eca-75c4-4e22-8177-eb62e10186f4)

![image](https://github.com/longhaseng52/dbeaver/assets/123133969/b262d60f-8745-4cf3-8282-c3d022ac1af8)

hwany7seo commented on plugins/org.jkiss.dbeaver.ext.cubrid/src/org/jkiss/dbeaver/ext/cubrid/model/meta/CubridMetaModel.java:
```diff
@@ -149,4 +205,226 @@ public GenericTableBase createTableOrViewImpl(
+        String description = JDBCUtils.safeGetString(dbResult, "comment");
+        String tableName = JDBCUtils.safeGetString(dbResult, "target_class_name");
+        DBRProgressMonitor monitor = dbResult.getSession().getProgressMonitor();
+        CubridTable cubridTable = null;
```
DBAs and DBA members can create triggers in different schemas. Therefore, cubridTable can be null.
I think I need to get all the table information.

longhaseng52 replied:
Noted, I will check this.

hwany7seo commented:
> Dear @hwany7seo,
> 
> QA team just found out that the column order is not matching with the design. It is because I follow the column order based on Generic. So, Should I change it following our design?

I commented in JIRA as below. Please refer to the image.

[Comment]
We are changing the table column design for the following reasons. ( the 8th (Design Change (Column)) page in 'tool_4478_2023_08_01.pptx')

1. We need the Virtua tab again. (This is a request from a DBeaver reviewer.)
2. Sorts the same as Generic, and unique can be created from the index, so it is removed from the Table Colums information.

The design file is replaced from 'tool_4478_2023_08_01.pptx' to 'tool_4478_2023_08_01_v1.1.pptx'.

![image](https://github.com/longhaseng52/dbeaver/assets/68524199/384809d7-f34f-4f18-aa35-559b09e5d9a5)

hwany7seo commented:
Foreign keys are not displayed in 11.0 version. Could you please check if it reproduces the same?

longhaseng52 commented:
> Foreign keys are not displayed in 11.0 version. Could you please check if it reproduces the same?

Yes, mine are not displayed in version 11.0 too. I will fix this. Thanks!

longhaseng52 committed:
Fixed foreign keys don't display in server 11.0

longhaseng52 committed:
Fixed error when displaying trigger in different target schema

longhaseng52 committed:
Fixed error description of Java Procedures/Functions

longhaseng52 committed:
Reformat code
- Place { at the same line where the method is declared
- Place all private methods at the bottom of the class

hwany7seo commented on plugins/org.jkiss.dbeaver.ext.cubrid/src/org/jkiss/dbeaver/ext/cubrid/model/CubridTableColumn.java:
```diff
@@ -0,0 +1,101 @@
+        if (dbResult != null) {
+            typeName = JDBCUtils.safeGetString(dbResult, "Type");
+            setName(JDBCUtils.safeGetString(dbResult, "Field"));
+            setFullTypeName(typeName);
```
ENUM type are displayed normally, but an exception still occurs.

I want the types to be displayed as is, as shown below.
ENUM, JSON
e.g) ENUM('A', 'B', 'C', 'D'), JSON('{"type": "string"}')

longhaseng52 replied:
Dear @hwany7seo 
I just want to verify, the screenshot below is what you want to display, right?
![image](https://github.com/longhaseng52/dbeaver/assets/123133969/b8250982-a012-4559-b099-24e3cdd83d38)

hwany7seo replied:
Yes, that's it.
Thanks @longhaseng52

longhaseng52 committed:
Fixed incorrect display dataype

hwany7seo commented:
@longhaseng52 
Last time I didn't display the foreign key in 11.0, now I don't display it in 11.2. Could you please check if it reproduces the same?

longhaseng52 committed:
Fixed foreign keys don't display in server 11.2

longhaseng52 commented:
> @longhaseng52 Last time I didn't display the foreign key in 11.0, now I don't display it in 11.2. Could you please check if it reproduces the same?

Dear @hwany7seo 

My mistake, I have just fixed it. Now, it works on both 11.0 and 11.2 version.

longhaseng52 committed:
Added annotations
